freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -170,12 +170,20 @@ public void cleanupByCluster(VaultCleanupRequest request, String accountId) thro
                 throw new DeleteException("Cluster CRN is required");
             }
             MDCBuilder.addResourceCrnToMdcContext(request.getClusterCrn());
-            VaultPathBuilder b = new VaultPathBuilder()
-                    .withAccountId(accountId)
-                    .withEnvironmentCrn(request.getEnvironmentCrn())
-                    .withClusterCrn(request.getClusterCrn());
-            recursivelyCleanupVault(b.withSubType(PRINCIPAL_SUB_TYPE).build());
-            recursivelyCleanupVault(b.withSubType(KEYTAB_SUB_TYPE).build());
+            cleanupSecrets(request.getEnvironmentCrn(), request.getClusterCrn(), accountId);
+        } catch (DeleteException e) {
+            throw e;
+        } catch (Exception e) {
+            LOGGER.error("Cleanup cluster failed " + e.getLocalizedMessage(), e);
+            throw new DeleteException("Failed to cleanup");
```

Comment:
I think it would make sense to add at least the message of the exception here

Reply:
Done